File: CharacterNexus/ClientApp/src/types/BonusAdjustment.ts
import { BonusCondition } from "./BonusCondition";

export interface BonusAdjustment {
    origin?: string;
    type: string;
    name: string;
    value: number;
    conditions?: BonusCondition[];
}

export interface BonusAdjustments extends Array<BonusAdjustment> {}

File: Utility/Models/BonusAdjustment.cs
using System;
using System.Collections.Generic;

namespace Utility
{
    public class BonusAdjustment
    {
        public string Type { get; set; } = "";
        public string Name { get; set; } = "";
        public int Value { get; set; } = 0;
        public List<BonusCondition>? Conditions { get; set; } = new List<BonusCondition>();
    }
}